src/GRA.Domain.Service/ActivityService.cs

```diff
@@ -220,7 +220,7 @@ int pointsEarned
                 // add the book if one was supplied
                 if (book != null && !string.IsNullOrWhiteSpace(book.Title))
                 {
-                    activityLogResult.BookId = await AddBookAsync(GetActiveUserId(), book);
+                    activityLogResult.BookId = (await AddBookAsync(GetActiveUserId(), book)).Data;
```

Comment:
If `AddBookAsync` goes badly here I think this de-reference is going to be an issue, it probably makes more sense to put `await AddBookAsync` into a variable and then do the assignment from that variable.